src/services/getEditsForFileRename.ts

```diff
@@ -1,70 +1,143 @@
+    type MovedFilePathUpdater = (sourceFile: SourceFile, importText: string) => string | undefined;
+    function getMovedFilePathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string): MovedFilePathUpdater {
+        const relativeNewDirToOldDir = getRelativePathFromDirectory(toDirectory(newFileOrDirPath), toDirectory(oldFileOrDirPath), /*ignoreCase*/ false);
+
+        if (relativeNewDirToOldDir === ".") return () => undefined;
+
+        return (sourceFile, importText) =>
+            !pathIsRelative(importText) ||
+                importIsInternalToDirectory(oldFileOrDirPath, sourceFile.fileName, importText) ||
+                importIsInternalToDirectory(newFileOrDirPath, sourceFile.fileName, importText)
+                ? undefined
+                : combineNormal(relativeNewDirToOldDir, importText);
+    }
+
+    function toDirectory(path: string): string {
+        return isAnySupportedFileExtension(path) ? getDirectoryPath(path) : path;
+    }
+
+    function importIsInternalToDirectory(oldDir: string, sourceFilePath: string, importText: string): boolean {
+        return !!removeDirectoryPrefixFromPath(oldDir, combineNormal(getDirectoryPath(sourceFilePath), importText));
+    }
+
+    /**
+     * @param oldFullPath Absolute path to a failed lookup location
+     * @param oldRelPath Actual import text
+     */
+    type PathUpdater = (oldFullPath: string, oldRelPath: string, isImport: boolean) => string | undefined;
+    function getPathUpdater(oldFileOrDirPath: string, newFileOrDirPath: string, host: LanguageServiceHost): PathUpdater {
         // Get the relative path from old to new location, and append it on to the end of imports and normalize.
-        const rel = getRelativePathFromFile(oldFilePath, newFilePath, createGetCanonicalFileName(hostUsesCaseSensitiveFileNames(host)));
-        return oldPath => {
-            if (!pathIsRelative(oldPath)) return;
-            return ensurePathIsNonModuleName(normalizePath(combinePaths(getDirectoryPath(oldPath), rel)));
+        const rel = getRelativePathFromFile(oldFileOrDirPath, newFileOrDirPath, createGetCanonicalFileName(hostUsesCaseSensitiveFileNames(host)));
+        return (fullPath, importText, isImport) => {
+            if (fullPath === oldFileOrDirPath) {
+                const newImportText = pathIsRelative(importText)
+                    ? combinePathsSafe(tryRemoveIndexOrPackage(fullPath) && !endsWithSomeIndex(importText) ? importText : getDirectoryPath(importText), rel)
+                    : newFileOrDirPath;
+                return isImport ? pathToImportPath(newImportText) : newImportText;
+            }
+            else {
+                // e.g., Importing from "/old/a/b", suffix is "/a/b", and we'll leave that part alone.
+                const relToOld = removeDirectoryPrefixFromPath(oldFileOrDirPath, fullPath);
+                if (relToOld === undefined) return undefined;
+                const suffix = isImport ? pathToImportPath(relToOld) : relToOld;

```

Comment:
moving folder ```/a/b/ ``` to folder say ```/a/c```  would need to move "resolvedFile name" of` ```import "foo"``` say ```/a/b/node_modules/foo/index.d.ts``` will make suffix as : "node_modules/foo" ? anmd newPrefix as "/a/c" and resulting import would be "/a/c/node_modules/foo" ?

Reply:
This function will only be called for imports from outside the moved directory, which won't resolve to `node_modules` in the moved directory unless it was an explicit relative import from "./b/node_modules/foo".